### PR TITLE
run crawler daily

### PR DESCRIPTION
**Pull request by daroczig**

### CODE REVIEW

File: .github/workflows/crawl.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      - name: Run SC Crawler
+        run: sc-crawler pull --connection-string sqlite:///$FILE --include-vendor aws
+      - name: Upload if changed
+        run: |
+          LHASH=`md5sum $FILE | cut -c -32`
+          RHASH=`aws s3api head-object --bucket $BUCKET --key $FILE --query "Metadata.md5" --output=text`
+          if [ "$LHASH" != "$RHASH" ]; then
```
Comment (bra-fsn) [inline .github/workflows/crawl.yaml:37]: Does sqlite produce the same output (given the same inputs)?

**Review by bra-fsn — APPROVED** [review]

File: .github/workflows/crawl.yaml (hunk 1 above)
Comment (daroczig) [reply .github/workflows/crawl.yaml:37]: almost :facepalm: 

image

Comment (daroczig) [reply .github/workflows/crawl.yaml:37]: here's the diff:

image

Comment (daroczig) [reply .github/workflows/crawl.yaml:37]: image

Comment (daroczig) [reply .github/workflows/crawl.yaml:37]: I think I can fix that :crossed_fingers:

Comment (bra-fsn) [reply .github/workflows/crawl.yaml:37]: I think it's pretty fragile to hash a database based on its binary representation. Many things can change, eg. they introduce a UUID etc. BTW, this approach also makes it impossible to include a timestamp (like when we could see an instance for the first time).
What could be done here is to hash on its contents. Of course we'll need to make sure it's consistent.
Problem statement [here](https://sqlite.org/forum/info/2e5883207f4ec282). :)

Comment (daroczig) [reply .github/workflows/crawl.yaml:37]: > Problem statement [here](https://sqlite.org/forum/info/2e5883207f4ec282). :)

Yeah, that is clear, and I think I can handle it as per above :crossed_fingers: 

> I think it's pretty fragile to hash a database based on its binary representation.

It is .. although it seems the same problem stands with a text dump .. so I feel pretty OK about it for now.

> this approach also makes it impossible to include a timestamp (like when we could see an instance for the first time).

IMO that's fine, as this SQLite DB is not to have a timestamp ... this is just a "most recent snapshot". The timestamp will go to the SCD version.

> Many things can change, eg. they introduce a UUID etc.

Yeah, good points, but IMO this DB should be reproducible, so I'm not worried about that.

> What could be done here is to hash on its contents.

I think that is much more relevant when it comes to the SCD table. For this "most recent snapshot", we "just" need to make sure that the DB is reproducible and the binary representation is deterministic.

Let's sync if you disagree, might be easier in a live call.

Comment (daroczig) [reply .github/workflows/crawl.yaml:37]: but I'm looking into hashing the content as well, as it will be useful for the SCD DB anyway :+1:

Comment (bra-fsn) [reply .github/workflows/crawl.yaml:37]: > > Many things can change, eg. they introduce a UUID etc.
> 
> Yeah, good points, but IMO this DB should be reproducible, so I'm not worried about that.

To make this clear, I haven't through about embedding a UUID into the DB by ourselves (via an INSERT etc), but Sqlite itself.
Like on each creation, it writes a unique identifier into the file.

Comment (daroczig) [reply .github/workflows/crawl.yaml:37]: db content hashing https://github.com/SpareCores/sc-crawler/pull/3

Comment (daroczig) [reply .github/workflows/crawl.yaml:37]: updated as per above

**Pull request merged**